### PR TITLE
[CI] bump pytorch to 2.0.1

### PR DESCRIPTION
**Pull request by johnnv1**
The required tests will need to be updated after this is merged. In addition, I think the tutorials test CI can be part of the required tests.

https://github.com/pytorch/pytorch/releases/tag/v2.0.1

### CODE REVIEW

**Comment by johnnv1** [comment]
For some reason, the 2.0.1 still failing the onnx export  -- same from 2.0 https://github.com/kornia/kornia/pull/2272#issuecomment-1492628216

**Review by edgarriba — APPROVED** [review]

**Review by edgarriba — APPROVED** [review]
All this we should report to PyTorch

File: .github/workflows/scheduled_test_cpu_half.yml [hunk 1]
```diff
@@ -20,6 +20,6 @@ jobs:
     uses: ./.github/workflows/tests.yml
     with:
       os: ${{ matrix.os }}
-      python-version: '["3.8", , "3.10"]' # "3.9"
-      pytorch-version:  '["2.0.0"]' # '["1.9.1", "1.10.2", "1.11.0", "1.12.1", "1.13.1"]'
+      python-version: '["3.8", , "3.10"]' # "3.7", "3.9"
```
Comment (johnnv1) [inline .github/workflows/scheduled_test_cpu_half.yml:23]:
```suggestion
      python-version: '["3.8", , "3.10"]' # "3.9"
```

**Review by edgarriba — COMMENTED** [review]
tests crashing

**Comment by johnnv1** [comment]
> tests crashing

i think are the same as always because of 1.9.1

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@johnnv1 Shoulw we target this for #2476 ?

**Comment by johnnv1** [comment]
> @johnnv1 Shoulw we target this for #2476 ?

i think this is fine to merge, the falling tests are unrelated with 2.0.1

**Pull request merged**